File: test_dev/osm2gmns_test_dev.cpp
//
// Created by Jiawei Lu on 2/17/23.
//

#include <exception>
#include <filesystem>
#include <iostream>
#include <memory>

#include "functions.h"
#include "io.h"
#include "networks.h"

int main(int /*argc*/, char* /*argv*/[]) {
  try {
    // const auto map_folder = std::filesystem::path("test_dev/maps/yuba");
    const auto map_folder = std::filesystem::path("test_dev/maps/imperial");

    std::unique_ptr<Network> network = getNetFromFile(map_folder / "map.osm.pbf", false);

    std::cout << "writing network\n";
    outputNetToCSV(*network, map_folder);

    std::cout << "done\n";
  } catch (const std::exception& e) {
    std::cerr << e.what() << '\n';
    return 1;
  }
  return 0;
}

File: test_dev/osm2gmns_test_dev.py
import os
import sys
parent_dir = os.path.abspath(os.path.join(os.path.dirname(__file__), '..'))
sys.path.append(parent_dir)

import osm2gmns as og

map_folder = r'test_dev/maps/yuba'
net = og.getNetFromFile(os.path.join(map_folder, 'map.osm.pbf'))
og.outputNetToCSV(net, map_folder)

